lib/pbench/test/functional/agent/cli/commands/test_config.py

```diff
@@ -0,0 +1,25 @@
+import os
+import pathlib
+import pytest
+
+
+def test_pbench_run(monkeypatch, tmpdir, agent_config, pbench_run):  # noqa F811
+    """Use pbench_run value from pbench-agent.cfg file"""
+    monkeypatch.delenv("pbench_run")
+    env_pbench_run = os.environ.get("pbench_run")
+    command = ["pbench-list-tools"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert exitcode == 0
+    assert env_pbench_run is None
+    assert pathlib.Path(f"{tmpdir}/var/lib/pbench-agent") == pbench_run
+
+
+def test_pbench_run_dir_existence(monkeypatch, agent_config, pbench_run):
+    """Verify the existence of a non-existing pbench_run directory
+    which results in expected failure of the test"""
+    monkeypatch.setenv("pbench_run", f"{pbench_run}/test")
+    command = ["pbench-list-tools"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert exitcode == 1
+    assert b"[ERROR] pbench run directory does not exist," in out
+    assert b"" == err
```

Comment:
It strikes me that we should have a third test which is just like these two tests combined:  it should pull in the `pbench_run` fixture and then use `monkeypatch` to override the environment variable, setting to a _valid_ directory *other than* the one from `pbench_run` (e.g., doing what `test_pbench_run_dir_existence` does but then _creating_ it), and then make sure that value of `BaseCommand.pbench_run` ends up matching the test's choice and not the config file's choice...although, the exact details of how to do that escape me....  😈

Reply:
Hey, @webbnh can you take a look at the implemented 3rd test. I believe there is some requirement for `Context` value, which I am feeling would be pretty difficult to retrieve or duplicate. Is it the right direction I am moving in?

Reply:
See my [comment](https://github.com/distributed-system-analysis/pbench/pull/2606#discussion_r803063752) below.